### PR TITLE
Allow two delete device records

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Addresses https://github.com/brave/brave-browser/issues/2817

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
This is more for DCHECK sanity. Impossible to test on release builds.
`npm run test -- brave_unit_tests --filter=BraveSyncServiceTest.OnDeleteDevice`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_sync/sync_devices.cc [hunk 1]
```diff
@@ -163,17 +158,14 @@ void SyncDevices::Merge(const SyncDevice& device,
...
-      //DCHECK(existing_device != nullptr);
       DCHECK(existing_it != std::end(devices_));
-      //*existing_device = device;
       *existing_it = device;
       *actually_merged = true;
       break;
     }
     case jslib_const::kActionDelete: {
-      //DCHECK(existing_device != nullptr);
-      DCHECK(existing_it != std::end(devices_));
```
Comment (darkdh) [inline components/brave_sync/sync_devices.cc]: DCHECK should only be triggered in debug build and what QA used for test is release build.
This doesn't solve the issue.

Comment (AlexeyBarabash) [reply components/brave_sync/sync_devices.cc]: Thanks for review. So the issue I address here https://github.com/brave/browser-android-tabs/issues/933 is fixed by another PR completely, will close issue. For this PR will create a new issue.

**Comment by AlexeyBarabash** [comment]
Rebased to current master.

**Comment by AlexeyBarabash** [comment]
Rebased to current master again and fixed lint issues.
@darkdh could you approve it again?

**Pull request merged**